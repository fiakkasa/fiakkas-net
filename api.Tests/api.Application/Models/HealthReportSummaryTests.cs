using Microsoft.Extensions.Diagnostics.HealthChecks;

namespace api.Application.Models.Tests;

public class HealthReportSummaryTests
{
    [Fact]
    public void Should_Convert_HealthReport_To_HealthReportSummary()
    {
        var duration = TimeSpan.FromSeconds(1);
        var healthReport = new HealthReport(
            new Dictionary<string, HealthReportEntry>
            {
                ["Key"] = new HealthReportEntry(
                    HealthStatus.Healthy, 
                    default, 
                    duration, 
                    default, 
                    default
                )
            },
            duration
        );

        var result = (HealthReportSummary)healthReport;

        result.Status.Should().Be(healthReport.Status);
        result.TotalDuration.Should().Be(healthReport.TotalDuration);
        result.Entries.Should().NotBeNull();
        result.Entries.Should().HaveCount(1);
        result.Entries.Should().ContainKey("Key");
        result.Entries!["Key"].Status.Should().Be(healthReport.Entries["Key"].Status);
        result.Entries!["Key"].Duration.Should().Be(healthReport.Entries["Key"].Duration);

        result.MatchSnapshot();
    }
}
