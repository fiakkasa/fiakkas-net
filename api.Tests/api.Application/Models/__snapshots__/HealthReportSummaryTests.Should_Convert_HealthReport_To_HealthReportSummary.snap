{
  "Status": "Healthy",
  "TotalDuration": "00:00:01",
  "Entries": {
    "Key": {
      "Description": null,
      "Duration": "00:00:01",
      "Status": "Healthy",
      "Tags": []
    }
  }
}
